### PR TITLE
Add higher-order spherical utility meshes

### PR DESCRIPTION
**Pull request by dorugeber**
Tests are fairly trivial, since we don't yet assemble anything on such meshes!

### CODE REVIEW

**Comment by miklos1** [comment]
Fine.

File: firedrake/utility_meshes.py [hunk 1]
```diff
@@ -523,8 +526,12 @@ def IcosahedralSphereMesh(radius, refinement_level=0, reorder=None):
 
     :kwarg refinement_level: optional number of refinements (0 is an
         icosahedron).
+    :kwarg degree: polynomial degree of coordinate space (defaults
+        to 1: flat triangles)
     :kwarg reorder: (optional), should the mesh be reordered?
     """
+    if degree < 1:
+        raise RuntimeError("Mesh coordinate degree must be at least 1")
```
Comment (kynan) [inline firedrake/utility_meshes.py]: Could be a `ValueError` instead

File: firedrake/utility_meshes.py [hunk 2]
```diff
@@ -713,17 +728,22 @@ def coordinates_on_panel(panel_num, X, Y, Z):
...
     :arg radius: The radius of the sphere to approximate.
     :kwarg refinement_level: optional number of refinements (0 is a cube).
+    :kwarg degree: polynomial degree of coordinate space (defaults
+        to 1: bilinear quads)
     :kwarg reorder: (optional), should the mesh be reordered?
     :kwarg use_dmplex_refinement: (optional), use dmplex to apply
         the refinement.
     """
+    if degree < 1:
+        raise RuntimeError("Mesh coordinate degree must be at least 1")
```
Comment (kynan) [inline firedrake/utility_meshes.py]: Could be a `ValueError` instead

**Comment by kynan** [comment]
I can believe this.

File: firedrake/utility_meshes.py (hunk 1 above)
Comment (dorugeber) [inline firedrake/utility_meshes.py]: Ah, fine.  Before writing this, I'd grepped for other `Error`s that we raised in this file, and found
```
    if length < 0:
        raise RuntimeError("Requested mesh has negative length")
```
so I went with RuntimeError.  Will change.

File: firedrake/utility_meshes.py (hunk 2 above)
Comment (dorugeber) [inline firedrake/utility_meshes.py]: Deetto

**Comment by miklos1** [comment]
There is an outstanding linting issue.

**Comment by dorugeber** [comment]
Will merge when Travis goes green.

**Pull request merged**